src/topic/TopicMessageSubmitTransaction.js

```diff
@@ -163,9 +163,19 @@ export default class TopicMessageSubmitTransaction extends Transaction {
      */
     setMessage(message) {
         this._requireNotFrozen();
-        this._message =
-            typeof message === "string" ? utf8.encode(message) : message;
-
+        if (message == undefined) {
+            throw new Error(
+                "TopicSubmitMessageTransaction.setMessage() given undefined message."
+            );
+        }
+        if (typeof message === "string" || message instanceof Uint8Array) {
+            this._message =
+                typeof message === "string" ? utf8.encode(message) : message;
+        } else {
+            throw new Error(
+                "TopicSubmitMessageTransaction.setMessage() given non string/Uint8Array message."
+            );
+        }
```

Comment:
Since you're already made an condition to exit early with an error I think it'd be a good idea to be consistent and make another condition to exit early if the types don't match.
```suggestion
        if (typeof message !== "string" && !(message instanceof Uint8Array)) {
            throw new Error(
                "TopicSubmitMessageTransaction.setMessage() given non string/Uint8Array message."
            );
        }

        this._message =
            typeof message === "string" ? utf8.encode(message) : message;
```